src/macaron/repo_finder/repo_finder.py

```diff
@@ -148,3 +149,72 @@ def to_repo_path(purl: PackageURL, available_domains: list[str]) -> str | None:
             fragment="",
         )
     )
+
+
+def find_source(purl_string: str, repo: str | None) -> bool:
+    """Perform repo and commit finding for a passed PURL, or commit finding for a passed PURL and repo.
+
+    Parameters
+    ----------
+    purl_string: str
+        The PURL string of the target.
+    repo: str | None
+        The optional repository path.
+
+    Returns
+    -------
+    bool
+        True if the source was found.
+    """
+    try:
+        purl = PackageURL.from_string(purl_string)
+    except ValueError as error:
+        logger.error("Could not parse PURL: %s", error)
+        return False
+
+    found_repo = repo
+    if not repo:
+        logger.debug("Searching for repo of PURL: %s", purl)
+        found_repo = find_repo(purl)
+
+    if not found_repo:
+        logger.error("Could not find repo for PURL: %s", purl)
+        return False
+
+    # Disable other loggers for cleaner output.
+    analyzer_logger = logging.getLogger("macaron.slsa_analyzer.analyzer")
+    analyzer_logger.disabled = True
+    git_logger = logging.getLogger("macaron.slsa_analyzer.git_url")
+    git_logger.disabled = True
+
+    # Prepare the repo.
+    logger.debug("Preparing repo: %s", found_repo)
+    # Importing here to avoid cyclic import problem.
+    from macaron.slsa_analyzer.analyzer import Analyzer  # pylint: disable=import-outside-toplevel, cyclic-import
+
+    analyzer = Analyzer(global_config.output_path, global_config.build_log_path)
+    repo_dir = os.path.join(analyzer.output_path, analyzer.GIT_REPOS_DIR)
+    git_obj = analyzer.prepare_repo(repo_dir, found_repo, "", "", purl)
```

Comment:
Looking again at the implementation of this `prepare_repo` function, it only uses `self` at a few locations to run static methods:
- https://github.com/oracle/macaron/blob/2d3a64a1b3af25d3dcc025f21d6018af0a15eb8a/src/macaron/slsa_analyzer/analyzer.py#L867
- https://github.com/oracle/macaron/blob/2d3a64a1b3af25d3dcc025f21d6018af0a15eb8a/src/macaron/slsa_analyzer/analyzer.py#L878
- https://github.com/oracle/macaron/blob/2d3a64a1b3af25d3dcc025f21d6018af0a15eb8a/src/macaron/slsa_analyzer/analyzer.py#L913
Do you think it would be worth while to refactor this method to another module to avoid having to silent the pylint `cyclic-import` error?